### PR TITLE
Cleanup, Add Check, and Write InitialOwner to Json

### PR DESCRIPTION
**Pull request by stevennevins**

### CODE REVIEW

File: script/M2_Deploy.s.sol [hunk 1]
```diff
@@ -34,12 +38,13 @@ contract Deployer_M2 is ExistingDeploymentParser {
...
-        address oracleInitialOwner = executorMultisig;
-        uint256 initialThreshold = stdJson.readUint(config_data, ".oracleInitialization.threshold");
+        oracleInitialOwner = executorMultisig;
+        initialThreshold = stdJson.readUint(config_data, ".oracleInitialization.threshold");
         bytes memory oracleSignerListRaw = stdJson.parseRaw(config_data, ".oracleInitialization.signers");
-        address[] memory initialOracleSigners = abi.decode(oracleSignerListRaw, (address[]));
+        initialOracleSigners = abi.decode(oracleSignerListRaw, (address[]));
 
         require(initialThreshold <= initialOracleSigners.length, "invalid initialThreshold");
+        require(initialThreshold >= 1, "invalid initialThreshold");
```
Comment (gpsanant) [inline script/M2_Deploy.s.sol:47]: We don't need the beacon chain oracle file actually, this is deployed by Succinct.

We do need all of the contracts in this script: https://github.com/Layr-Labs/eigenlayer-contracts/blob/m2-upgrade-script/script/upgrade/GoerliM2Upgrade.s.sol

Could you add them?

File: script/output/M2_deployment_data.json [hunk 2]
```diff
@@ -1,9 +1,11 @@
 {
   "addresses": {
-    "beaconChainOracle": "0x90193C961A926261B756D1E5bb255e67ff9498A1"
```
Comment (gpsanant) [inline script/output/M2_deployment_data.json:3]: .gitignore

File: script/M2_Deploy.s.sol (hunk 1 above)
Comment (stevennevins) [reply script/M2_Deploy.s.sol:47]: 🫡 added

**Comment by stevennevins** [comment]
@gpsanant added the beacon chain oracle from succinct

File: script/output/M2_deployment_data.json [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
```
Comment (gpsanant) [inline script/output/M2_deployment_data.json:3]: still not ignored

Comment (stevennevins) [reply script/output/M2_deployment_data.json:3]: copied the .gitignore line for M1 and realized it doesn't work lol

updated

should I fix the one for the M1 and remove that as well?

Comment (gpsanant) [reply script/output/M2_deployment_data.json:3]: ah, yes i guess. not sure if we're keeping them here for clarity? doubt it

**Review by gpsanant — APPROVED** [review]
lgtm

**Pull request merged**